src/mapgen.cpp

```diff
@@ -3621,7 +3657,8 @@ class jmapgen_nested : public jmapgen_piece
         }
         const weighted_int_list<mapgen_value<nested_mapgen_id>> &get_entries(
         const mapgendata &dat ) const {
-            if( neighbor_oters.test( dat ) && neighbor_joins.test( dat ) && neighbor_flags.test( dat ) ) {
+            if( neighbor_oters.test( dat ) && neighbor_joins.test( dat ) && neighbor_flags.test( dat ) &&
+                neighbor_flags_any.test( dat ) ) {
```

Comment:
`neighbor_flags && neighbor_flags_any` is redundant: `x AND (x OR y)` simplifies to `x AND x `which simplifies to `x`.

Reply:
This isn't redundant they're entirely separate checks that can contain different flags for different directions and be combined eg `x AND (y or z)`

Reply:
Apologies, I must have misunderstood the change intent.

Reply:
You can look at nest4's placement in the documentation I changed for an example where it needs to be &&